### PR TITLE
feat: add universal module

### LINKED ISSUE #244: add universal module
It should be quite easy to add a universal module, kinda like `edk2` or `u-root` is at the moment - it just runs user-defined command is a docker container.

This way people could use `firmware-action` even for yet-unsupported projects, or even some non-firmware projects and steps.

Instead of:
```json
"build_command": "cmd"
```

it would use:
```json
"build_commands": [
  "cmd1",
  "cmd2",
]
```

### PR DESCRIPTION
**Pull request by AtomicFS**
fixes #244 

I just copied `u-root` and renamed stuff, removed unnecessary things, and changed the build command into slice instead of single string.

### CODE REVIEW

File: action/recipes/universal.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"dagger.io/dagger"
+	"github.com/9elements/firmware-action/action/container"
+)
+
+// UniversalSpecific is used to store data specific to universal
```
Comment (MDr164) [inline action/recipes/universal.go:15]: I know there comments have just been slightly adapted from the u-root module but they read really weird now. Maybe something like this?
```suggestion
// UniversalSpecific is used to store data specific to the universal command module
```

File: action/recipes/universal.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+type UniversalSpecific struct {
+	// Specifies build commands to execute inside container
+	BuildCommands []string `json:"build_commands" validate:"required"`
+}
+
+// ANCHOR_END: UniversalSpecific
+
+// ANCHOR: UniversalOpts
+
+// UniversalOpts is used to store all data needed to run universal
```
Comment (MDr164) [inline action/recipes/universal.go:26]: Same here. Maybe:
```suggestion
// UniversalOpts is used to store all data needed to run universal commands
```

File: action/recipes/universal.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+func (opts UniversalOpts) GetDepends() []string {
+	return opts.Depends
+}
+
+// GetArtifacts returns list of wanted artifacts from container
+func (opts UniversalOpts) GetArtifacts() *[]container.Artifacts {
+	return opts.CommonOpts.GetArtifacts()
+}
+
+// buildFirmware builds universal
```
Comment (MDr164) [inline action/recipes/universal.go:51]: Also here but not sure what to change it to

**Review by MDr164 — COMMENTED** [review]
Code-wise looking good, left some comments about the docs/comments being a bit odd on this one

File: action/recipes/universal.go (hunk 1 above)
Comment (AtomicFS) [reply action/recipes/universal.go:15]: `universal command module` :D sounds like NASA project :D

But yeah, you are right, they sound odd, but I had no idea what to change it to. `universal module` just sounds a but weird.

**Review by MDr164 — APPROVED** [review]

**Pull request merged**